### DISCUSSION TITLE
Error on click in Related videos

### DISCUSSION THREAD

**Discussion by anthonycassol** [LonelyCpp/react-native-youtube-iframe#133] [Q&A]
Hii. I'm having a problem clicking on related videos. Is it possible to disable them?

Error:
TypeError: undefined is not an object (evaluating 'request.mainDocumentURL.startsWith').

Thanks Everyone.

**Comment by LonelyCpp** [comment]
huh, seems like a bug. I will look into it.

you can override the behaviour for now like this - 

```jsx
<YoutubePlayer

  webViewProps={{ 
    onShouldStartLoadWithRequest: (request) => {
        // your custom handling here
        // return false here to stop a unwanted redirection (like clicking on a related video)
    } 
  }}

/>
```

info on how it works: [documentation](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md#onshouldstartloadwithrequest)

**Reply by anthonycassol** [reply]
error whitout webViewProps
![image](https://user-images.githubusercontent.com/28939625/115404642-e786d500-a1c3-11eb-9342-899c5a03ae9b.png)

this is the code with webViwProps
![image](https://user-images.githubusercontent.com/28939625/115403731-1486b800-a1c3-11eb-85e2-113f2abeee0b.png)

now when i click it opens the webview, no longer the error
![image](https://user-images.githubusercontent.com/28939625/115404465-bdcdae00-a1c3-11eb-96a9-eff3427661f5.png)

**Reply by LonelyCpp** [reply]
you'll have to write some logic here like this - 

```jsx
 webViewProps={{ 
    onShouldStartLoadWithRequest: (request) => {
        // your custom handling here
        // return false here to stop a unwanted redirection (like clicking on a related video)
    } 
  }}
```

1. inspect `request.url` (android) or `request.mainDocumentUrl` (ios)
2. if the URL has the same video ID as your video, then return true (continue to load)
3. if it does not, you'll have to return `false` extract the new video ID from the URL and give that id to the player
